pandas/_libs/tslibs/strptime.pyx

```diff
@@ -173,6 +175,10 @@ def array_strptime(
                 check_dts_bounds(&dts)
             result_timezone[i] = val.tzinfo
             continue
+        elif PyDate_Check(val):
+            iresult[i] = pydate_to_dt64(val, &dts)
+            check_dts_bounds(&dts)
+            continue
```

Comment:
same as

https://github.com/pandas-dev/pandas/blob/7c0278e533b8cf19a8b6bd367bcb26a06d08ab9e/pandas/_libs/tslib.pyx#L544-L547

Reply:
do we need to worry about found_naive?  (i guess we dont currently in the tslib code?)

Reply:
Not sure we need to, `datetime.date`s can't have a timezone, so perhaps it's fine to just parse?

Reply:
OK to punt on this for now